examples/nix-shell/shell.ncl

```diff
@@ -0,0 +1,20 @@
+{
+  inputs = {
+    hello = { input = "nixpkgs" }
+  }
+
+  , output = fun { pkgs, nix, .. } =>
```

Comment:
I think it would be nice to make it explicit that this `pkgs` is derived from the `inputs` field. I remember we didn’t want to call it `inputs` because it could lead to confusing errors since there’s already an `inputs` in scope, but maybe we could rename the `inputs` field to `inputs_spec` or something similar to prevent this.

(besides, `pkgs` isn’t restricted to packages, so it would be nice for it to have a broader name)

Reply:
Thought hard about a better name, `inputs_decl`, `inputs_requested`, etc. but went with `inputs_spec` in the end. Not great, but the less badest :slightly_smiling_face: